alica_engine/include/engine/AlicaEngine.h

```diff
@@ -111,10 +112,21 @@ class AlicaEngine
     essentials::IdentifierConstPtr getIDFromBytes(const uint8_t *idBytes, int idSize, uint8_t type = essentials::Identifier::UUID_TYPE);
     essentials::IdentifierConstPtr generateID(std::size_t size);
 
+    void reload(const YAML::Node& config);
+    const YAML::Node& getConfig() const;
+    void subscribe(std::function<void(const YAML::Node& config)> reloadFunction);
+
+    /**
+     * Call reload() of all subscribed components. Each component does reload using the
+     * updated config.
+     */
+    void reloadConfig(const YAML::Node& config);
+
 private:
     void setStepEngine(bool stepEngine);
     // WARNING: Initialization order dependencies!
     // Please do not change the declaration order of members.
+    std::vector<std::function<void(const YAML::Node& config)>> _reloadFunctions;
```

Comment:
we can have a more meaningful name like _configChangeSubscribers

Reply:
Since it is a listener pattern, _configChangeListenerCBs could also work.